src/pages/dataTableUtilities/reducer/reducerMethods.js

```diff
@@ -252,3 +262,160 @@ export const sortData = (state, action) => {
+  let addedItem;
+
+  database.write(() => {
+    if (pageObject.hasItem(item)) return;
+    addedItem = createRecord(database, addedItemType, pageObject, item);
+  });
+
+  if (addedItem) return { ...state, data: [addedItem, ...data], modalKey: '' };
+
+  return { ...state, modalKey: '' };
+};
+
+/**
+ * Reducer helper method for editing a non-setter field on a
+ * realm object. Should be called by more explicit reducer
+ * methods and not exported to avoid misuse. To create a reducer
+ * method for setting a simple value for a realm object, create a
+ * more explicit reducer method with logic specific to that field
+ * contained, and call this method. See editTheirRef.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'editPageObject', value, field, pageObjectType }
+ */
+const editPageObject = (state, action) => {
+  const { database, pageObject } = state;
+  const { value, pageObjectType, field } = action;
+
+  database.write(() => {
+    pageObject[field] = value;
+    database.save(pageObjectType, pageObject);
+  });
+
+  return { ...state, modalKey: '' };
+};
+
+/**
+ * Edits the passed pageObject 'theirRef' field with the value supplied.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'editPageObject', value, pageObjectType }
+ */
+export const editTheirRef = (state, action) => {
+  const { pageObject } = state;
+  const { value } = action;
+
+  const { theirRef } = pageObject;
+
+  if (theirRef !== value) return { ...editPageObject(state, { ...action, field: 'theirRef' }) };
+  return state;
+};
+
+/**
+ * Edits the passed pageObject 'comment' field with the value supplied.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ * Action: { type: 'editPageObject', value, pageObjectType }
+ */
+export const editComment = (state, action) => {
+  const { pageObject } = state;
+  const { value } = action;
+
+  const { comment } = pageObject;
+
+  if (comment !== value) return { ...editPageObject(state, { ...action, field: 'comment' }) };
+  return state;
+};
+
+/**
+ * Deletes the selected RequisitionItems or
+ * TransactionItems held in state. Where each selected
+ * item is indicated by DataState[rowKey].isSelected.
+ *
+ * @param {Object} state  The current state
+ * @param {Object} action The action to act upon
+ */
+export const deleteItemsById = (state, action) => {
+  const { database, pageObject, dataState, hasSelection, backingData } = state;
+  const { pageObjectType } = action;
+
+  if (!hasSelection) return state;
+
+  const itemsIds = Array.from(dataState.keys()).filter(rowKey => dataState.get(rowKey).isSelected);
+
+  database.write(() => {
+    pageObject.removeItemsById(database, itemsIds);
+    database.save(pageObjectType, pageObject);
+  });
+
+  const newDataState = new Map();
+  const newData = backingData.slice();
```

Comment:
Testing I found a problem with this:
![image](https://user-images.githubusercontent.com/7684221/63568098-47c90c00-c5c8-11e9-9768-2650de990b91.png)
Basically need to reapply the sorting? Or splice the the same rows out of `data` instead of slicing again but that's pretty sketchy (e.g. the delete failed on DB, but rows disappear in UI)

Reply:
14a209667c787a312149128358aca51bb6213f12

improved the deletion speed a loooot!